web/ui/src/Pools/Pool.js

```diff
@@ -0,0 +1,81 @@
+(function () {
+    'use strict';
+
+    // share angular services outside of angular context
+    let POOL_PERMISSIONS, $q, resourcesFactory;
+
+    controlplane.factory('Pool', PoolFactory);
+
+    // TODO - this is a dummy class till the fo'real one is in
+    // TODO - import real Host
+    function Host(model){
+        this.id = model.ID;
+        this.name = model.Name;
+        this.model = Object.freeze(model);
+    }
+
+    // Pool object constructor takes a pool object (backend pool object)
+    // and wraps it with extra functionality and info
+    class Pool{
+    
+        constructor(pool){
+            this.id = pool.ID;
+            this.model = Object.freeze(pool);
+            this.hosts = [];
+            this.calculatePermissions();
+            this.touch();
+        }
+
+        // mark services updated to trigger render via $watch
+        touch() {
+            this.lastUpdate = new Date().getTime();
+        }
+
+        calculatePermissions(){
+            // build a list of permissions
+            // this pool has
+            // NOTE: permissions include description
+            // and friendly label for the UI to display
+            let val = this.model.Permissions;
+            this.permissions = [];
+            POOL_PERMISSIONS.forEach(perm => {
+                if((val & perm.position) !== 0){
+                    this.permissions.push(perm);
+                }
+            });
+        }
+
+        fetchHosts(force){
+            let deferred = $q.defer();
+            if (this.hosts && !force) {
+                deferred.resolve();
+            }
+            // TODO - this is actually a v2 endpoint
+            // and should be on resourcesFactory.v2
+            resourcesFactory.getPoolHosts(this.id)
```

Comment:
This call to the v2 api endpoint (using the UI's v1 api object) will be replaced with the correct call in the upcoming commits for the 101 pools UI changes.